pennylane/pulse/rydberg_hamiltonian.py

```diff
@@ -0,0 +1,289 @@
+        interaction_coeff (float): Rydberg interaction constant in units: :math:`\text{MHz} \times \mu \text{m}^6`.
+            Defaults to :math:`862690 \times 2\pi \text{ MHz} \times \mu \text{m}^6`.
+
+    Returns:
+        RydbergHamiltonian: a :class:`~.ParametrizedHamiltonian` representing the atom interaction
+
+    **Example**
+
+    .. code-block:: python
+
+        atom_coordinates = [[0, 0], [0, 5], [5, 0], [10, 5], [5, 10], [10, 10]]
+        wires = [1, 5, 0, 2, 4, 3]
+        RH = qml.pulse.rydberg_interaction(atom_coordinates, wires=wires)
+
+    >>> RH
+    ParametrizedHamiltonian: terms=3
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit.jax", wires=6)
+
+        @qml.qnode(dev, interface="jax")
+        def circuit():
+            # Since the Hamiltonian is dependent only on the number and positions of Rydberg atoms, the evolution
+            # does not have any trainable parameters
+            qml.evolve(RH)([], t=[0, 10])
+            return qml.expval(qml.PauliZ(0))
+
+    >>> circuit()
+    Array(1., dtype=float32)
+    """
+    if wires is None:
+        wires = list(range(len(register)))
+    elif len(wires) != len(register):
+        raise ValueError("The length of the wires and the register must match.")
+
+    def rydberg_projector(wire: int) -> SProd:
+        """Returns the projector into the Rydberg state for the given wire.
+
+        Args:
+            wire (int): wire value
+
+        Returns:
+            SProd: projector into the Rydberg state
+        """
+        return qml.s_prod(1 / 2, (1 - qml.PauliZ(wire)))
+
+    coeffs = []
+    observables = []
+    for idx, (pos1, wire1) in enumerate(zip(register[:-1], wires[:-1])):
+        for pos2, wire2 in zip(register[(idx + 1) :], wires[(idx + 1) :]):
+            atom_distance = np.linalg.norm(qml.math.array(pos2) - pos1)
+            Vij = interaction_coeff / (abs(atom_distance) ** 6)  # van der Waals potential
+            coeffs.append(Vij)
+            observables.append(qml.prod(rydberg_projector(wire1), rydberg_projector(wire2)))
+
+    return RydbergHamiltonian(
+        coeffs, observables, register=register, interaction_coeff=interaction_coeff
+    )
+
+
+def rydberg_drive(rabi, detuning, phase, wires):
+    r"""Returns a :class:`ParametrizedHamiltonian` representing the action of a driving laser
+    field with the given rabi frequency, detuning and phase acting on the given wires
+
+    .. math::
+
+        \frac{1}{2} \Omega(t) \sum_i (\cos(\phi)\sigma_i^x - \sin(\phi)\sigma_i^y) -
+        \frac{1}{2} \delta(t) \sum_i \sigma_i^z
+
+    where :math:`\Omega`, :math:`\delta` and :math:`\phi` correspond to the rabi frequency, detuning
+    and phase of the laser, :math:`i` correspond to the wire index, and :math:`\sigma^\alpha` for
+    :math:`\alpha = x,y,z` are the Pauli matrices.
+
+    Args:
+        rabi (Union[float, Callable]): float or callable returning the frequency (in MHz) of a laser
```

Comment:
```suggestion
        rabi (Union[float, Callable]): float or callable returning the amplitude (in MHz) of a laser
```

How attached are we to the name `rabi` for this variable? Braket-AHS uses `amplitude`, and I think this is clearer. The term `rabi` to me is more strongly associated with frequency (and it doesn't help that its measured in Hz 😅 ), which makes it confusing when there is also a detuning.

Reply:
I changed it to amplitude. Lillian and I were also having some discussion on slack regarding what the units should be. Since they will be converted to rad/s during upload to braket, should we just use rad/s as the unit here as well?

Reply:
For numerical accuracy we will probably want to operate on MHz or GHz to avoid using very large and very small numbers. I know for transmons it makes sense to use GHz for freq./energy and ns for time, then everything is unit amplitude. How is that here?

Reply:
MHz is the relevant scale here